castle/cms/cron/_auto_publish_retract.py

```diff
@@ -0,0 +1,74 @@
+from AccessControl.SecurityManagement import newSecurityManager
+from DateTime import DateTime
+from plone import api
+from Products.CMFPlone.interfaces.siteroot import IPloneSiteRoot
+from tendo import singleton
+from zope.component.hooks import setSite
+
+import logging
+import transaction
+
+
+logger = logging.getLogger('castle.cms')
+
+
+def run_query(query_params):
+
+    catalog = api.portal.get_tool('portal_catalog')
+    content = catalog(**query_params)
+    
+    to_state = 'published' if query_params.get('effective') else 'private'
+    for brain in content:
+        obj = brain.getObject()
+        api.content.transition(
+            obj=obj, 
+            to_state=to_state
+        )
+        obj.setModificationDate()
+        obj.reindexObject(idxs=['modified'])
+    transaction.commit()
+
+
+def set_queries(site):
+    setSite(site)
+
+    # publish pending content with an effective date within 30 minutes of cron run
+    start = DateTime() - (30.0 / (24 * 60))
+    end = DateTime()
+    publish_query = {
+        'effective': {
+            'query': (start, end),
+            'range': 'min:max'
+        },
+        'review_state': ['pending', 'private']
+    }
+    run_query(publish_query)
```

Comment:
I wonder if this should be named to indicate that it has side effects, and does just get a result -- eg "run_transitions" or something along those lines?

Reply:
Noted, changed for clarity.